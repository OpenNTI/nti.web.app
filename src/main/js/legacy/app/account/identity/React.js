const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
