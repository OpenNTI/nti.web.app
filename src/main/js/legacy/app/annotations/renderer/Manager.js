var Ext = require('extjs');
var Anchors = require('../../../util/Anchors');
var Globals = require('../../../util/Globals');
var UtilLine = require('../../../util/Line');


module.exports = exports = Ext.define('NextThought.app.annotations.renderer.Manager', {
    events: new Ext.util.Observable(),
    registry: [],

    constructor: function(reader) {
		this.callParent();
		this.reader = reader;
		this.rendererSuspended = 0;
	},

    controlLineTmpl: Ext.DomHelper.createTemplate({ cls: 'controlContainer'}).compile(),
    isDebug: false,

    /**
	 * @constructor Inner class
	 */
	Bucket: function() {
		this.values = {};
		this.length = 0;

		this.free = function() {
			this.each(function(v, k, o) { delete o[k]; }, this);
			Ext.Object.each(this, function(k, o, me) { delete me[k]; });
		};

		this.put = function(o, key) {
			key = typeof key === 'number' ? key : guidGenerator();
			if (this.values[key]) {
				throw 'existing value';
			}
			this.values[key] = o;
			this.length += 1;
			return key;
		};

		this.each = function(cb, s) {
			Ext.Object.each(this.values, function(key, value, o) {
				Ext.callback(cb, s, [value, key, o]);
			});
		};

		this.get = function(key) {
			return this.values[key];
		};

		this.first = function() {
			var key;
			for (key in this.values) {
				if (this.values.hasOwnProperty(key)) {
					return this.get(key);
				}
			}
			return null;
		};
	},

    registerGutter: function(el, reader) {
		//TODO all this junk about prefixes should go away once we aren't using a singleton here...
		if (this.gutter) {
			console.warn('replacing exisiting gutter?', this.gutter);
		}
		this.gutter = el;

		//el.addCls('debug');

		el.controls = el.down('.controls');

		this.render();//renders wait for the gutter to exist
	},

    register: function(o) {
		this.registry.push(o);
		o.requestRender();
	},

    unregister: function(o) {
		var r = this.registry;
		if (r) {
			this.registry = Ext.Array.remove(r, o);
			if (this.registry.length === 0) {
				this.sorter = null;
			}
		}
	},

    getReader: function() {
		return this.reader;
	},

    buildSorter: function(prefix) {
		//Default sort will sort by lastModified
		return function(a, b) {
			var e = {get: Ext.emptyFn},
					c = 0, $a = (a.record || e).get('Last Modified'), $b = (b.record || e).get('Last Modified');
			if ($a !== $b) {
				c = $a < $b ? 1 : -1;
			}

			return c;
		};
	},

    clearBuckets: function() {
		function clear(d) {
			while (d && d.firstChild) {
				d.removeChild(d.firstChild);
			}
		}

		var g = this.gutter;

		if (this.gutter.down('.active')) {
			this.selectedLine = this.gutter.down('.active').getAttribute('data-line');
		}
		clear(g.controls.dom);

		if (this.buckets) {
			this.buckets.free();
		}
		this.buckets = new this.Bucket();
	},

    getBucket: function(line) {
		//console.debug('prefix:'+prefix, line);
		if (line < 0) {
			//bad line, don't render:
			if (this.isDebug) {
				console.error('Annotation cannot be rendered in gutter');
			}
			return null;
		}

		var c = this.buckets,
			lineTolerance = 40,
			b = c ? c.get(line) : null,
			a = Math.round(line - (lineTolerance / 2)), z = Math.round(line + (lineTolerance / 2));

		if (!b && c) {
			c.each(function(value, key) {
				var keyNum = parseInt(key, 10);
				if (keyNum >= a && keyNum <= z) {
					b = value;
					return false;
				}
				return true;
			});

			if (!b) {
				b = new this.Bucket();
				b.line = line;
				c.put(b, line);
			}
		}
		return b;
	},

    layoutBuckets: function() {
		var g = this.gutter,
			b = this.buckets,
			cT = this.controlLineTmpl;


		b.each(function(line, y) {
			var count = 0;

			line.controls = line.controls || cT.append(g.controls, [], true);
			line.controls.setStyle('top', y + 'px');
			line.controls.set({'data-line': y});

			line.each(function(o) {
				if (o.isNote) {
					++count;
				}
			});
			if (count > 0) {
				Ext.DomHelper.overwrite(line.controls, {cls: 'count', html: count});
			}

			if (!line.controls.first()) {
				line.controls.remove();
				delete line.controls;
			}
		});
	},

    suspend: function() {
		this.rendererSuspended++;
	},

    resume: function() {
		this.rendererSuspended--;
		if (this.rendererSuspended === 0 && this.renderOnResume) {
			delete this.renderOnResume;
			this.render();
		}
	},

    render: function() {
		var me = this, containers = {}, renderedCount = 0,
			cleanContent, rootContainerId,
			cloned, descs = [], cids = [], doc = null, selectedEl;

		if (me.rendering) {
			console.warn('Render called while rendering...');
			me.events.on('finish', Ext.bind(me.render, me, arguments), me, {single: true});
			return;
		}
		if (this.rendererSuspended > 0) {
			this.renderOnResume = true;
			return;
		}
		if (!me.gutter) {
			console.error('no gutter');
			me.events.fireEvent('rendering');
			me.events.fireEvent('finish', 0);
			return;
		}

		if (!me.registry) {
			me.events.fireEvent('rendering');
			me.events.fireEvent('finish', 0);
			return;//nothing to do
		}

		me.rendering = true;
		me.events.fireEvent('rendering');
		console.log('Rendering annotations');
		if (console.time) {
			console.time('Annotation render loop');
		}
		Ext.suspendLayouts();

		me.registry = Ext.Array.unique(me.registry);

		me.sorter = me.sorter || me.buildSorter();
		if (me.sorter) {
			Ext.Array.sort(me.registry, me.sorter);
		}

		me.clearBuckets();

		cloned = Ext.Array.clone(me.registry);
		Ext.each(cloned, function(o) {
			var desc = o.getRecordField ? o.getRecordField('applicableRange') : null,
					cid = o.getRecordField ? o.getRecordField('ContainerId') : null;
			if (o.doc) {
				doc = o.doc;
			}
			descs.push(desc);
			cids.push(cid);
		});

		if (cloned && cloned.length > 0 && doc) {
			cleanContent = me.getReader().getCleanContent();
			rootContainerId = Anchors.rootContainerIdFromDocument(doc);
			Anchors.preresolveLocatorInfo(descs, doc, cleanContent, cids, rootContainerId);
			Ext.each(cloned, function(o) {
				var y, b, c;
				try {
					if (!o.isVisible) {
						return;
					}

					c = containers[o.getContainerId()] = (containers[o.getContainerId()] || []);

					if (o.isNote) {
						c.push(o);
					}

					y = o.render();
					if (y !== NextThought.app.annotations.Base.HIDDEN) {
						b = me.getBucket(Math.ceil(y));
						if (b) {
							o.getRecord().set('line', b.line);
							renderedCount++;
							b.put(o);
						}
					}
				}
				catch (e) {
					console.error(o.$className, Globals.getError(e));
				}
			});
			console[renderedCount === cloned.length ? 'log' : 'warn']('Rendered ' + renderedCount + '/' + cloned.length + ' annotations');
			me.layoutBuckets();
		}

		me.rendering = false;
		Ext.resumeLayouts(true);
		me.events.fireEvent('finish', renderedCount);
		if (console.timeEnd) {
			console.timeEnd('Annotation render loop');
		}
		if (this.selectedLine) {
			selectedEl = this.gutter.down('[data-line=' + this.selectedLine + ']');
			if (selectedEl) {
				selectedEl.addCls('active');
			}
		}
	}
}, function() {
	var me = this,
			fn = this.prototype.render,
			timerId;

	me.prototype.render = function() {
		var callerScope = this;
		if (timerId) {
			clearTimeout(timerId);
		}
		timerId = setTimeout(function() {
			fn.call(callerScope);
		}, 100);
	};
});
