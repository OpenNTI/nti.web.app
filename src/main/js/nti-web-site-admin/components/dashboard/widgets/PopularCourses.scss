@import "~nti-style-common/all.scss";

.dashboard-list-widget {
	margin: 5px;
	width: 450px;
	height: 300px;
	box-shadow: 0 1px 2px 1px #ccc;
	color: #555;
	font-weight: 300;

	figure.loading {
		position: relative;
		top: 20%;
	}

	.header {
		display: flex;
		position: relative;
		margin: 15px 10px 20px 15px;

		.title {
			font-size: 22px;
		}

		.pager {
			display: flex;
			position: absolute;
			right: 0;

			.page-control {
				border: solid 1px #e5e5e5;
				width: 34px;
				height: 30px;
				padding-left: 10px;
				padding-top: 4px;
				color: #888;
				cursor: pointer;

				&.disabled {
					color: #ccc;
				}

				&.previous {
					border-radius: 30px 0 0 30px;
				}

				&.next {
					border-radius: 0 30px 30px 0;
				}
			}
		}
	}

	.items-container {
		margin: 15px;

		.items-header {
			display: flex;
			position: relative;
			color: #b8b8b8;
			font-size: 14px;
			margin-bottom: 10px;

			.column-header.value {
				position: absolute;
				right: 0;
			}
		}

		.item {
			display: flex;
			position: relative;
			margin-bottom: 10px;

			.rank {
				font-size: 12px;
				color: #b8b8b8;
				padding-top: 10px;
				margin-right: 10px;
				min-width: 20px;
			}

			.item-image {
				width: 40px;
				height: 40px;
				margin-right: 15px;
				background-size: cover;
			}

			.info {
				padding-top: 3px;
				font-weight: 400;

				.name {
					font-size: 14px;
				}

				.description {
					font-size: 12px;
					color: #b8b8b8;
				}
			}

			.value {
				position: absolute;
				right: 0;
				font-size: 20px;
			}
		}
	}
}
