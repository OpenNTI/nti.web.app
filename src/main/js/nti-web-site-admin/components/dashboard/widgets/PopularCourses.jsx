import './PopularCourses.scss';
import React from 'react';
import PropTypes from 'prop-types';
import { scoped } from '@nti/lib-locale';
import { Presentation, Loading } from '@nti/web-commons';
import { getService } from '@nti/web-client';
import cx from 'classnames';

import { getString } from 'legacy/util/Localization';

const LABELS = {
	title: getString(
		'NextThought.view.courseware.assessment.admin.dashboard.widget.PopularCourses'
	),
	name: 'Course Rank',
	value: 'Students',
	noItems: 'No courses found',
};

const t = scoped(
	'nti-web-site-admins.components.dashboard.widgets.popularcourses',
	LABELS
);
const PAGE_SIZE = 4;

class Item extends React.Component {
	static propTypes = {
		item: PropTypes.object.isRequired,
	};

	renderRank() {
		const { item } = this.props;

		return <div className="rank">{item.rank}.</div>;
	}

	renderImg() {
		const { item } = this.props;

		return (
			<Presentation.AssetBackground
				className="item-image"
				contentPackage={item}
				type="landing"
			/>
		);
	}

	renderInfo() {
		const { item } = this.props;

		return (
			<div className="info">
				<div className="name">{item.name}</div>
				<div className="description">{item.description}</div>
			</div>
		);
	}

	renderValue() {
		const { item } = this.props;

		return <div className="value">{item.value}</div>;
	}

	render() {
		return (
			<div className="item">
				{this.renderRank()}
				{this.renderImg()}
				{this.renderInfo()}
				{this.renderValue()}
			</div>
		);
	}
}

export default class PopularCourses extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			loading: true,
			pageNumber: 0,
		};
	}

	componentDidMount() {
		this.setState({ items: [] }, () => {
			this.loadData();
		});
	}

	loadData(link) {
		const { pageNumber } = this.state;

		const batchStart = pageNumber * PAGE_SIZE;

		this.setState({ loading: true });

		getService().then(service => {
			const collection = service.getCollection('Courses', 'Catalog');
			const popularLink = collection && collection.getLink('Popular');

			if (popularLink) {
				const getBatchLink = link
					? link
					: popularLink +
					  '?batchSize=' +
					  PAGE_SIZE +
					  '&batchStart=' +
					  batchStart;

				service
					.getBatch(getBatchLink)
					.then(results => {
						this.setState({
							loading: false,
							totalPages: Math.ceil(results.Total / PAGE_SIZE),
							prevLink: results.getLink('batch-prev'),
							nextLink: results.getLink('batch-next'),
							items: results.Items.map((x, i) => {
								return {
									...x,
									name: x.Title,
									description: x.ProviderUniqueID,
									rank: batchStart + i + 1,
									value: x.TotalEnrolledCount,
								};
							}),
						});
					})
					.catch(resp => {
						this.setState({
							loading: false,
							items: [],
						});
					});
			} else {
				// no popular link, set items to empty
				this.setState({
					loading: false,
					items: [],
				});
			}
		});
	}

	onPrevious = () => {
		const { pageNumber, prevLink } = this.state;

		if (pageNumber === 0) {
			return;
		}

		this.setState(
			{
				pageNumber: pageNumber - 1,
			},
			() => {
				this.loadData(prevLink);
			}
		);
	};

	onNext = () => {
		const { pageNumber, totalPages, nextLink } = this.state;

		if (pageNumber >= totalPages) {
			return;
		}

		this.setState(
			{
				pageNumber: pageNumber + 1,
			},
			() => {
				this.loadData(nextLink);
			}
		);
	};

	renderHeader() {
		// disabled state for previous includes a check where pageNumber is 0.  This way, if we get to that last "phantom" page where we have no
		// links (prev or next), we can still go back to the previous real page
		const prevClassName = cx('page-control', 'previous', {
			disabled:
				this.state.loading ||
				(!this.state.prevLink && this.state.pageNumber === 0),
		});
		const nextClassName = cx('page-control', 'next', {
			disabled: this.state.loading || !this.state.nextLink,
		});

		return (
			<div className="header">
				<div className="title">{t('title')}</div>
				<div className="pager">
					<div className={prevClassName} onClick={this.onPrevious}>
						<i className="icon-chevron-left" />
					</div>
					<div className={nextClassName} onClick={this.onNext}>
						<i className="icon-chevron-right" />
					</div>
				</div>
			</div>
		);
	}

	renderItem = (item, index) => {
		return <Item key={item.name + index} item={item} />;
	};

	renderItems() {
		const { items, loading } = this.state;

		if (loading) {
			return <Loading.Mask />;
		} else if (items && items.length === 0) {
			return <div className="no-items">{t('noItems')}</div>;
		}

		return (
			<div className="items-container">
				<div className="items-header">
					<div className="column-header name">{t('name')}</div>
					<div className="column-header value">{t('value')}</div>
				</div>
				<div className="items">
					{(items || []).map(this.renderItem)}
				</div>
			</div>
		);
	}

	render() {
		return (
			<div className="dashboard-list-widget popular-courses">
				{this.renderHeader()}
				{this.renderItems()}
			</div>
		);
	}
}
