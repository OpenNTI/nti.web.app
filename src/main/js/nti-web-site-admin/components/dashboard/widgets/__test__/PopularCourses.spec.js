import React from 'react';
import renderer from 'react-test-renderer';
import { TestUtils } from '@nti/web-client';

import PopularCourses from '../PopularCourses';

const { tearDownTestClient, setupTestClient } = TestUtils;

const getMockService = (numberOfCourses, hasBatchPrev) => {
	return {
		getCollection: () => {
			return {
				hasLink: () => true,
				getLink: () => 'mockLink'
			};
		},
		getWorkspace: () => {
			return {};
		},
		getBatch: () => {
			let Items = [];

			for(let i = 0; i < numberOfCourses; i++) {
				const index = i + 1;

				Items.push({
					Title: 'course' + index,
					ProviderUniqueID: 'COURSE' + index,
					TotalEnrolledCount: 100 - index
				});
			}

			return Promise.resolve({
				Items,
				Total: Items.length,
				getLink: (link) => {
					if(link === 'batch-prev') {
						return hasBatchPrev;
					}
					else if(link === 'batch-next') {
						return numberOfCourses > 4;
					}

					return false;
				}
			});
		}
	};
};

const onBefore = (numberOfCourses, hasBatchPrev) => {
	jest.useFakeTimers();
	setupTestClient(getMockService(numberOfCourses, hasBatchPrev));
};

const onAfter = () => {
	tearDownTestClient();
};

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

/* eslint-env jest */
describe('Site admin dashboard widget popular courses (5 items, no previous)', () => {
	beforeEach(() => onBefore(5));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<PopularCourses/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Site admin dashboard widget popular courses (3 items, has previous)', () => {
	beforeEach(() => onBefore(3, true));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<PopularCourses/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Site admin dashboard widget popular courses (no items, no previous)', () => {
	beforeEach(() => onBefore(0));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<PopularCourses/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
