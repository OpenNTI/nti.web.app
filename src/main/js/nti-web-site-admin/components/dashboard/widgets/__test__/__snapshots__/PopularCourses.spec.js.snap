// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin dashboard widget popular courses (3 items, has previous) Basic render test 1`] = `
<div
  className="dashboard-list-widget popular-courses"
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      NextThought.view.courseware.assessment.admin.dashboard.widget.PopularCourses
    </div>
    <div
      className="pager"
    >
      <div
        className="page-control previous"
        onClick={[Function]}
      >
        <i
          className="icon-chevron-left"
        />
      </div>
      <div
        className="page-control next disabled"
        onClick={[Function]}
      >
        <i
          className="icon-chevron-right"
        />
      </div>
    </div>
  </div>
  <div
    className="items-container"
  >
    <div
      className="items-header"
    >
      <div
        className="column-header name"
      >
        Course Rank
      </div>
      <div
        className="column-header value"
      >
        Students
      </div>
    </div>
    <div
      className="items"
    >
      <div
        className="item"
      >
        <div
          className="rank"
        >
          1
          .
        </div>
        <div
          className="item-image"
          onError={[Function]}
          src="contentpackage-landing-232x170.png?t=0"
          style={
            Object {
              "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
            }
          }
        />
        <div
          className="info"
        >
          <div
            className="name"
          >
            course1
          </div>
          <div
            className="description"
          >
            COURSE1
          </div>
        </div>
        <div
          className="value"
        >
          99
        </div>
      </div>
      <div
        className="item"
      >
        <div
          className="rank"
        >
          2
          .
        </div>
        <div
          className="item-image"
          onError={[Function]}
          src="contentpackage-landing-232x170.png?t=0"
          style={
            Object {
              "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
            }
          }
        />
        <div
          className="info"
        >
          <div
            className="name"
          >
            course2
          </div>
          <div
            className="description"
          >
            COURSE2
          </div>
        </div>
        <div
          className="value"
        >
          98
        </div>
      </div>
      <div
        className="item"
      >
        <div
          className="rank"
        >
          3
          .
        </div>
        <div
          className="item-image"
          onError={[Function]}
          src="contentpackage-landing-232x170.png?t=0"
          style={
            Object {
              "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
            }
          }
        />
        <div
          className="info"
        >
          <div
            className="name"
          >
            course3
          </div>
          <div
            className="description"
          >
            COURSE3
          </div>
        </div>
        <div
          className="value"
        >
          97
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Site admin dashboard widget popular courses (5 items, no previous) Basic render test 1`] = `
<div
  className="dashboard-list-widget popular-courses"
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      NextThought.view.courseware.assessment.admin.dashboard.widget.PopularCourses
    </div>
    <div
      className="pager"
    >
      <div
        className="page-control previous disabled"
        onClick={[Function]}
      >
        <i
          className="icon-chevron-left"
        />
      </div>
      <div
        className="page-control next"
        onClick={[Function]}
      >
        <i
          className="icon-chevron-right"
        />
      </div>
    </div>
  </div>
  <div
    className="items-container"
  >
    <div
      className="items-header"
    >
      <div
        className="column-header name"
      >
        Course Rank
      </div>
      <div
        className="column-header value"
      >
        Students
      </div>
    </div>
    <div
      className="items"
    >
      <div
        className="item"
      >
        <div
          className="rank"
        >
          1
          .
        </div>
        <div
          className="item-image"
          onError={[Function]}
          src="contentpackage-landing-232x170.png?t=0"
          style={
            Object {
              "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
            }
          }
        />
        <div
          className="info"
        >
          <div
            className="name"
          >
            course1
          </div>
          <div
            className="description"
          >
            COURSE1
          </div>
        </div>
        <div
          className="value"
        >
          99
        </div>
      </div>
      <div
        className="item"
      >
        <div
          className="rank"
        >
          2
          .
        </div>
        <div
          className="item-image"
          onError={[Function]}
          src="contentpackage-landing-232x170.png?t=0"
          style={
            Object {
              "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
            }
          }
        />
        <div
          className="info"
        >
          <div
            className="name"
          >
            course2
          </div>
          <div
            className="description"
          >
            COURSE2
          </div>
        </div>
        <div
          className="value"
        >
          98
        </div>
      </div>
      <div
        className="item"
      >
        <div
          className="rank"
        >
          3
          .
        </div>
        <div
          className="item-image"
          onError={[Function]}
          src="contentpackage-landing-232x170.png?t=0"
          style={
            Object {
              "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
            }
          }
        />
        <div
          className="info"
        >
          <div
            className="name"
          >
            course3
          </div>
          <div
            className="description"
          >
            COURSE3
          </div>
        </div>
        <div
          className="value"
        >
          97
        </div>
      </div>
      <div
        className="item"
      >
        <div
          className="rank"
        >
          4
          .
        </div>
        <div
          className="item-image"
          onError={[Function]}
          src="contentpackage-landing-232x170.png?t=0"
          style={
            Object {
              "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
            }
          }
        />
        <div
          className="info"
        >
          <div
            className="name"
          >
            course4
          </div>
          <div
            className="description"
          >
            COURSE4
          </div>
        </div>
        <div
          className="value"
        >
          96
        </div>
      </div>
      <div
        className="item"
      >
        <div
          className="rank"
        >
          5
          .
        </div>
        <div
          className="item-image"
          onError={[Function]}
          src="contentpackage-landing-232x170.png?t=0"
          style={
            Object {
              "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
            }
          }
        />
        <div
          className="info"
        >
          <div
            className="name"
          >
            course5
          </div>
          <div
            className="description"
          >
            COURSE5
          </div>
        </div>
        <div
          className="value"
        >
          95
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Site admin dashboard widget popular courses (no items, no previous) Basic render test 1`] = `
<div
  className="dashboard-list-widget popular-courses"
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      NextThought.view.courseware.assessment.admin.dashboard.widget.PopularCourses
    </div>
    <div
      className="pager"
    >
      <div
        className="page-control previous disabled"
        onClick={[Function]}
      >
        <i
          className="icon-chevron-left"
        />
      </div>
      <div
        className="page-control next disabled"
        onClick={[Function]}
      >
        <i
          className="icon-chevron-right"
        />
      </div>
    </div>
  </div>
  <div
    className="no-items"
  >
    No courses found
  </div>
</div>
`;
